Engine/src/Application.bf

```diff
@@ -0,0 +1,79 @@
+using SteelEngine.Window;
+using SteelEngine.Events;
+
+namespace SteelEngine
+{
+	/*
+	Usage:
+
+	class Sandbox : Application
+	{
+		public override void OnInit()
+		{
+			base.OnInit();
+		}
+	
+		public override void OnCleanup()
+		{
+			base.OnCleanup();
+		}	
+	}
+
+	static void Main()
+	{
+		var app = scope Sandbox();
+		app.Run();
+	}
+
+	*/
+	public abstract class Application
+	{
+		private bool mIsRunning = false;
+
+		private Window mWindow ~ delete _;
```

Comment:
Given this [vote on the Discord](https://discordapp.com/channels/738020284954574939/738497936119431169/738814211651797023), these should be prefixed with `_` rather than `m`.